### PR TITLE
Feature/include local export file

### PR DESCRIPTION
**Pull request by decadef20**
## Description

Add a short description of:
- what changes you made, 
   #421 
- why you made them, and 
- any other context that you think might be helpful for someone to better understand what is contained in this pull request. 

This sort of information is useful for people reviewing the code, as well as anyone from the future trying to understand why changes were made or why a bug started happening.

Please help review react and vue framework test snap. If passed,  I'll fully support other frameworks.

### CODE REVIEW

**Comment by decadef20** [comment]
@samijaber  help review this PR

File: packages/core/src/__tests__/vue.test.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ describe('Vue', () => {
     const output = componentToVue()({ component, path });
     expect(output).toMatchSnapshot();
   });
+  return;
```
Comment (samijaber) [inline packages/core/src/__tests__/vue.test.ts:34]: This line delete the snapshots of all the tests below.
```suggestion
```

Comment (decadef20) [reply packages/core/src/__tests__/vue.test.ts:34]: Sorry. This is debug code. Forget to delete.

Comment (decadef20) [reply packages/core/src/__tests__/vue.test.ts]: Fixed.

**Review by PatrickJS — APPROVED** [review]

File: packages/core/src/parsers/jsx.ts [hunk 2]
```diff
@@ -853,6 +878,42 @@ export function parseJsx(
...
+                  isFunction = babel.types.isFunction(
+                    (node.declaration as any).declarations[0].init,
+                  );
+                }
+                if (babel.types.isVariableDeclaration(node)) {
+                  name = (node.declarations as any)[0].id.name;
+                  isFunction = babel.types.isFunction(
+                    (node.declarations as any)[0].init,
+                  );
+                }
```
Comment (samijaber) [inline packages/core/src/parsers/jsx.ts:903]: This allows a bunch of nodes that will cause crashes. We should add checks for `isIdentifier` to the inner node


```suggestion
                if (
                  babel.types.isExportNamedDeclaration(node) &&
                  babel.types.isVariableDeclaration(node.declaration) &&
                  babel.types.isIdentifier(node.declaration.declarations[0].id)
                ) {
                  name = node.declaration.declarations[0].id.name;
                  isFunction = babel.types.isFunction(
                    node.declaration.declarations[0].init,
                  );
                } else if (
                  babel.types.isVariableDeclaration(node) &&
                  babel.types.isIdentifier(node.declarations[0].id)
                ) {
                  name = node.declarations[0].id.name;
                  isFunction = babel.types.isFunction(
                    node.declarations[0].init,
                  );
                }
```

**Review by samijaber — COMMENTED** [review]
The most important issue to fix is the one about the `jsx` parser: removing the `any` casts and adding more node type checks to avoid crashes

File: packages/core/src/generators/vue.ts [hunk 3]
```diff
@@ -490,6 +509,13 @@ export const componentToVue =
       );
     }
 
+    if (localVarAsFunc.length) {
+      functionsString = functionsString.replace(
+        /}\s*$/,
+        `${localVarAsFunc.join(',')}}`,
+      );
+    }
```
[the remark this answers is not in the document]
Comment (decadef20) [reply packages/core/src/generators/vue.ts:517]: I have tried to modify `component.state` directly but failed to put it in right place.

The plugin looks great, I'll try it later.

File: packages/core/src/parsers/jsx.ts (hunk 2 above)
Comment (decadef20) [reply packages/core/src/parsers/jsx.ts:903]: I'll fix it later

File: packages/core/src/generators/vue.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (decadef20) [reply packages/core/src/generators/vue.ts:517]: I tried to use plugin, 
```bash
-   data: () => ({ name: "Steve", DEFAULT_VALUES }),
+   data: () => ({ name: "Steve", DEFAULT_VALUES: "DEFAULT_VALUES" }),
```
Obviously, it's not what we want .
Or we could copy value to data, like this 
```bash
-   data: () => ({ name: "Steve", DEFAULT_VALUES }),
+   data: () => ({ name: "Steve", DEFAULT_VALUES: { name: "Steve",} }),
```

**Pull request merged**